pkg/osutil/fileutil.go

```diff
@@ -81,3 +83,29 @@ func WriteTempFile(data []byte) (string, error) {
 	f.Close()
 	return f.Name(), nil
 }
+
+// GrepFiles returns the list of files (relative to root) that include target.
+// If ext is not empty, the files will be filtered by the extension.
+// The function assumes that the files are not too big and may fit in memory.
+func GrepFiles(root, ext string, target []byte) ([]string, error) {
+	var ret []string
+	err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() || filepath.Ext(path) != ext {
+			return nil
+		}
+		content, err := os.ReadFile(path)
+		if err != nil {
+			// Ignore errors.
```

Comment:
What errors do we expect?
It's better to start with stricter code, and then relax it when/if we know about concrete problems/errors we want to ignore.